File: webcam.rb
#!/usr/bin/env ruby
# webcam.rb
# encoding: UTF-8

require 'opencv'
require 'optparse'

options = { output: 'snapshot.jpg', device: 0, delay: 0 }

OptionParser.new do |opts|
  opts.banner = "Usage: webcam.rb [options]"
  opts.on('-o', '--output FILE', 'Выходной файл') { |v| options[:output] = v }
  opts.on('-d', '--device N', Integer, 'Индекс камеры') { |v| options[:device] = v }
  opts.on('-t', '--delay N', Integer, 'Задержка в секундах') { |v| options[:delay] = v }
  opts.on('-h', '--help', 'Справка') { puts opts; exit }
end.parse!

include OpenCV

capture = CvCapture.open(options[:device])
unless capture
  STDERR.puts "Ошибка: не удалось открыть камеру"
  exit 1
end

if options[:delay] > 0
  puts "Ожидание #{options[:delay]} секунд..."
  sleep options[:delay]
end

frame = capture.query
unless frame
  STDERR.puts "Ошибка: не удалось захватить кадр"
  exit 1
end

frame.save(options[:output])
puts "✅ Изображение сохранено в #{options[:output]}"

File: webcam.cs
// webcam.cs
using System;
using Emgu.CV;
using Emgu.CV.CvEnum;

class WebcamCapture
{
    static void Main(string[] args)
    {
        string output = "snapshot.jpg";
        int device = 0, delay = 0;

        for (int i = 0; i < args.Length; i++)
        {
            if (args[i] == "-o" && i+1 < args.Length) output = args[++i];
            else if (args[i] == "-d" && i+1 < args.Length) device = int.Parse(args[++i]);
            else if (args[i] == "-t" && i+1 < args.Length) delay = int.Parse(args[++i]);
            else if (args[i] == "-h" || args[i] == "--help")
            {
                Console.WriteLine("Usage: webcam [-o output] [-d device] [-t delay]");
                return;
            }
        }

        using (var capture = new VideoCapture(device))
        {
            if (!capture.IsOpened)
            {
                Console.Error.WriteLine("Ошибка: не удалось открыть камеру");
                Environment.Exit(1);
            }

            if (delay > 0)
            {
                Console.WriteLine($"Ожидание {delay} секунд...");
                System.Threading.Thread.Sleep(delay * 1000);
            }

            using (var frame = capture.QueryFrame())
            {
                if (frame == null)
                {
                    Console.Error.WriteLine("Ошибка: не удалось захватить кадр");
                    Environment.Exit(1);
                }
                frame.Save(output);
                Console.WriteLine($"✅ Изображение сохранено в {output}");
            }
        }
    }
}
